parser/parser.go

```diff
@@ -39,10 +39,22 @@ type Package struct {
 
 // ParsePackage parses the package in the given directory and returns it.
 func ParsePackage(directory string) (*Package, error) {
-	relDir, err := filepath.Rel(filepath.Join(build.Default.GOPATH, "src"), directory)
-	if err != nil {
-		return nil, fmt.Errorf("provided directory not under GOPATH (%s): %v",
-			build.Default.GOPATH, err)
+	var (
+		relDir string
+		inPath bool
+	)
+	for _, gopath := range filepath.SplitList(build.Default.GOPATH) {
+		srcPath := filepath.Join(gopath, "src")
+		if strings.HasPrefix(directory, srcPath) {
+			relDir = strings.TrimPrefix(directory, srcPath)
+			relDir = strings.TrimPrefix(relDir, "/")
+			inPath = true
+			break
+		}
+	}
```

Comment:
Consider simplifying the above code by using [`build.ImportDir`](https://godoc.org/go/build#ImportDir) with [`build.FindOnly`](https://godoc.org/go/build#FindOnly) flag:

```Go
p, err := build.ImportDir(directory, build.FindOnly)
if err != nil {
	return nil, err
}
// use p.ImportPath in place of of relDir
```